프론트, 백엔드 빌드하는 과정에서 에러가 발생했다.

## 버그 내용
> 버그에 대한 내용을 알려주세요.

## 버그 상황
> 어떤 상황에서 버그가 발생했는지 알려주세요.

### 프론트 빌드

```typescript

166         return { isSuccess: true, data: data.responses, type: ApiType.PRODUCT, code: res.status };
                                                                  ~~~~~~~

src/apis/user/product.ts:168:22 - error TS2304: Cannot find name 'AxiosError'.

168       .catch((error: AxiosError) => {
                         ~~~~~~~~~~

src/apis/user/product.ts:172:17 - error TS2304: Cannot find name 'ApiType'.

172           type: ApiType.PRODUCT,
                    ~~~~~~~

src/apis/user/product.ts:185:63 - error TS2304: Cannot find name 'ApiType'.

185         return { isSuccess: true, data: data.responses, type: ApiType.PRODUCT, code: res.status };
                                                                  ~~~~~~~

src/apis/user/product.ts:187:22 - error TS2304: Cannot find name 'AxiosError'.

187       .catch((error: AxiosError) => {
                         ~~~~~~~~~~

src/apis/user/product.ts:191:17 - error TS2304: Cannot find name 'ApiType'.

191           type: ApiType.PRODUCT,
                    ~~~~~~~
```

### 백엔드 빌드

- 백엔드에서 빌드하는 과정에서 DeclarationService 클래스에서 createDeclaration 메서드 부분에서 reportStatus 안에 있는 값을 ReportStatus enum 클래스에 있는 값과 일치하지 않아서 에러가 발생했습니다.
- 따라서 ReportStatus enum 클래스에 있는 값과 일치하도록 해당 부분 값을 수정했습니다.

#### 수정 전

```java
@Transactional(readOnly = true)
@Service
public class DeclarationService {

    private final UserRepository userRepository;
    private final ProductRepository productRepository;
    private final DeclarationRepository declarationRepository;

    public DeclarationService(UserRepository userRepository,
                              ProductRepository productRepository,
                              DeclarationRepository declarationRepository) {
        this.userRepository = userRepository;
        this.productRepository = productRepository;
        this.declarationRepository = declarationRepository;
    }
  ...
  private Declaration createDeclaration(User user, Product product ,DeclarationSaveRequest request) {
          Declaration newDeclaration = Declaration.builder()
                  .declarationCategory(request.getDeclarationCategory())
                  .content(request.getContent())
                  .reportStatus(Processing) // 에러 원인: PROCESSING으로 바꿔야함
                  .user(user)
                  .product(product)
                  .build();
          return newDeclaration;
      }
   ...
}     
```

#### 수정 후

```java
@Transactional(readOnly = true)
@Service
public class DeclarationService {

    private final UserRepository userRepository;
    private final ProductRepository productRepository;
    private final DeclarationRepository declarationRepository;

    public DeclarationService(UserRepository userRepository,
                              ProductRepository productRepository,
                              DeclarationRepository declarationRepository) {
        this.userRepository = userRepository;
        this.productRepository = productRepository;
        this.declarationRepository = declarationRepository;
    }
  ...
  private Declaration createDeclaration(User user, Product product ,DeclarationSaveRequest request) {
          Declaration newDeclaration = Declaration.builder()
                  .declarationCategory(request.getDeclarationCategory())
                  .content(request.getContent())
                  .reportStatus(PROCESSING) // 수정후
                  .user(user)
                  .product(product)
                  .build();
          return newDeclaration;
      }
  ...
}     
```




## 참고 자료
> 참고할 만한 내용 혹은 주소가 있다면 작성해 주세요.

--- a/backend/src/main/java/woorifisa/goodfriends/backend/declaration/application/DeclarationService.java
+++ b/backend/src/main/java/woorifisa/goodfriends/backend/declaration/application/DeclarationService.java
@@ -11,7 +11,7 @@ import woorifisa.goodfriends.backend.product.domain.ProductRepository;
 import woorifisa.goodfriends.backend.user.domain.User;
 import woorifisa.goodfriends.backend.user.domain.UserRepository;
 
-import static woorifisa.goodfriends.backend.declaration.domain.ReportStatus.Processing;
+import static woorifisa.goodfriends.backend.declaration.domain.ReportStatus.PROCESSING;
 
 @Transactional(readOnly = true)
 @Service
@@ -44,7 +44,7 @@ public class DeclarationService {
         Declaration newDeclaration = Declaration.builder()
                 .declarationCategory(request.getDeclarationCategory())
                 .content(request.getContent())
-                .reportStatus(Processing)
+                .reportStatus(PROCESSING)
                 .user(user)
                 .product(product)
                 .build();

--- a/frontend/src/apis/user/product.ts
+++ b/frontend/src/apis/user/product.ts
@@ -163,15 +163,17 @@ const productAPI = {
       })
       .then((res: AxiosResponse) => {
         const { data } = res;
-        return { isSuccess: true, data: data.responses, type: ApiType.PRODUCT, code: res.status };
+        return { isSuccess: true, data: data.responses,  code: res.status };
       })
-      .catch((error: AxiosError) => {
-        return {
-          isSuccess: false,
-          message: error.message,
-          type: ApiType.PRODUCT,
-          code: error.status
-        };
+      .catch((error) => {
+        if (error.response) {
+          return {
+            isSuccess: false,
+            message: error.response.data.message,
+            code: error.response.status
+          };
+        }
+        return { isSuccess: false, message: error.message, code: error.response.status };
       });
   },
   getSearchTitleProduct: (productCategory: string, keyword: string): Promise<IResultType<Array<IAllProduct>>> => {
@@ -182,15 +184,17 @@ const productAPI = {
       })
       .then((res: AxiosResponse) => {
         const { data } = res;
-        return { isSuccess: true, data: data.responses, type: ApiType.PRODUCT, code: res.status };
+        return { isSuccess: true, data: data.responses, code: res.status };
       })
-      .catch((error: AxiosError) => {
-        return {
-          isSuccess: false,
-          message: error.message,
-          type: ApiType.PRODUCT,
-          code: error.status
-        };
+      .catch((error) => {
+        if (error.response) {
+          return {
+            isSuccess: false,
+            message: error.response.data.message,
+            code: error.response.status
+          };
+        }
+        return { isSuccess: false, message: error.message, code: error.response.status };
       });
   }
 };

